tracer/test/Datadog.Trace.Tests/Propagators/SpanContextPropagatorFactoryTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿// <copyright file="SpanContextPropagatorFactoryTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Propagators;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Propagators;
+
+public class SpanContextPropagatorFactoryTests
+{
+    [Theory]
+    [InlineData("invalid")]
+    [InlineData("1, 2, 3")]
+    [InlineData("")]
+    [InlineData(null)]
+    public void InvalidContextPropagator(string headerStyles)
+    {
+        var propagators = SpanContextPropagatorFactory.GetPropagators<IContextExtractor>(headerStyles?.Split(','));
+
+        propagators.Should().NotBeNull().And.BeEmpty();
```

Comment:
Shouldn't we have a fallback here where no propagators are parsed? Or is that handled elsewhere?

Reply:
Although I think I'll let `SpanContextPropagatorFactory` return an empty enumerable and handle the default value in `TracerSettings`.

Reply:
On second thought, we've had requests from users that want to disable propagation / distributed tracing. Allowing an empty list of propagators would resolve this, although it would require using an invalid input, like `DD_TRACE_PROPAGATION_STYLE_INJECT=disabled`, which feels wonky.

Reply:
> Good point. For other parsed settings like integers and Booleans we fall back to the default value if they are not valid. Do you think we should do the same for these

That was my initial thought, ideally logging a warning that we're falling back.

>On second thought, we've had requests from users that want to disable propagation / distributed tracing. Allowing an empty list of propagators would resolve this

That's a very interesting idea, though it feels like it would be better to have a dedicated `DD_TRACE_PROPAGATION_ENABLED` flag instead. Especially as customers normally don't want to disable it _entirely_, they typically want to only enable/disable for specific domains. 

We may need to think about data streams monitoring too, which also injects headers. Not really related to w3c etc, but it is propagation _related_.

Reply:
That makes sense! However, I won't include that in this PR since that is a change from current behavior.

If any of the `DD_TRACE_PROPAGATION_STYLE[*]` settings are not set, they default to `"Datadog"` in `TracerSettings` and we split by commas. Later, in `TracerManagerFactory`, we create a propagator instance for each string in the setting, ignoring any unknown strings.

The default values will change in a separate PR, so I can tackle the change discussed here at that time.